software/common/test/frame_detector_test.cpp

```diff
@@ -0,0 +1,286 @@
+/* Copyright 2020-2022, RespiraWorks
```

Comment:
It is good to see many test cases here for such an important functionality.

If application counters were applied these could also be asserted here.